ginga/AstroImage.py

```diff
@@ -223,35 +198,29 @@ def get_data_size(self):
         return self.get_size()
 
     def get_header(self, create=True, include_primary_header=None):
-        try:
-            # By convention, the fits header is stored in a dictionary
-            # under the metadata keyword 'header'
-            hdr = self.metadata['header']
-
-            if include_primary_header is None:
-                include_primary_header = self.inherit_primary_header
-
-            if include_primary_header and self._primary_hdr is not None:
-                # Inherit PRIMARY header for display but keep metadata intact
-                displayhdr = AstroHeader()
-                for key in hdr.keyorder:
-                    card = hdr.get_card(key)
-                    bnch = displayhdr.__setitem__(card.key, card.value)
-                    bnch.comment = card.comment
-                for key in self._primary_hdr.keyorder:
-                    if key not in hdr:
-                        card = self._primary_hdr.get_card(key)
-                        bnch = displayhdr.__setitem__(card.key, card.value)
-                        bnch.comment = card.comment
-            else:
-                # Normal, separate header
-                displayhdr = hdr
-
-        except KeyError as e:
+        # By convention, the fits header is stored in a dictionary
+        # under the metadata keyword 'header'
+        if 'header' not in self.metadata:
             if not create:
-                raise e
+                # TODO: change to ValueError("No header found")
+                raise KeyError('header')
```

Comment:
Do you intent to address the TODO as part of this PR?

Reply:
No, that would change the exception type.  Will look at that for the follow-up PR.